mitodl/bootcamps-zendesk-theme
# Sync the bootcamps Zendesk theme with the upstream

### What are the relevant tickets?

[#3840](https://github.com/mitodl/hq/issues/3840) 


### Description (What does it do?)
Here is the Testing URL of re-synced theme: https://arbisoft8524.zendesk.com/hc/en-us

This PR syncs the Zendesk theme with the upstream. It maintains the existing design sync and incorporates updates and features from the Zendesk upstream theme.

### How should this be manually tested?

1. Homepage Changes:

- Visit the [old bootcamps Zendesk home page](https://mitbootcamps.zendesk.com/hc/en-us) and compare it with the [new bootcamps Zendesk home page](https://arbisoft8524.zendesk.com/hc/en-us).
- Ensure that the clear search feature is functional on multiple pages.
- Confirm that the search submit button on the homepage is enabled and working.

2. Submit a Request Page:

- Navigate to the [old submit a request page](https://mitbootcamps.zendesk.com/hc/en-us/requests/new) and compare it with the [new submit a request page](https://arbisoft8524.zendesk.com/hc/en-us/requests/new).

- Ensure that the layout and functionality of the new submit a request page match the expected changes and incorporate new changes if any.

3. "Sign In" link:
- Confirm that the "Sign In" link is not visible in the test environment and will not be present when deployed to production or verify that the "Sign In" link is disabled in the header for both desktop and mobile views. 

Overall Testing:

- Conduct general testing of the Zendesk theme to ensure that all existing features are functioning correctly after the sync with the upstream.
- Test any other specific features or functionalities mentioned in the changelog or observed during the sync process.

### Screenshots (if appropriate):

**Old bootcamps zendesk home page:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/7c65b156-46a6-4821-9052-c28224901f16)

**New bootcamps zendesk home page:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/1983d7dc-4f47-4922-999a-fcee0a39f58d)

**Old bootcamps zendesk categories page for articles:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/80567785-3ba0-492d-8d3e-989a9c43ce3b)

**New bootcamps zendesk categories page for articles:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/10278c2e-7a05-4454-9f48-df3ff88aa09f)

**Old submit a request page:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/72505852-2a9c-4392-b162-e180325939b9)

**New submit a request page:**
![image](https://github.com/mitodl/bootcamps-zendesk-theme/assets/41048311/b20365a2-49d3-4a28-b6cd-4348ba67bc21)


### Additional Context

-  While working on this, there was no workflow for auditing URLs using Lighthouse before. Now it exists but is not yet configured. Should we configure or let it go?

- For now, disabled/skipped some jobs that did not exist before and require a decision on whether to keep them disabled or not. ~~One job, "Lint commit messages," is intentionally disabled because its purpose is to check commit conventions from the base SHA to the new pull request SHA. However, in our case (of making a pull request - sync upstream with the master branch), where there are many old commits in copenhagen_theme that don't follow commit conventions before that job was introduced, this job will fail. We can re-enable it to enforce commit convention rules after the sync-upstream-master branch is merged with master.~~ After the strategy change, this will now rely on our commits made to this fork.
